### ISSUE TITLE
Infer `x: Final = "string"` as constant of type `Literal["string"]` rather than just constant `str`

### ISSUE THREAD

**Issue by zevisert** [microsoft/pylance-release#2174]
Typing a string as `Final` should infer that it's also a literal and that should be used in intellisense hovers. This is a feature from mypy.

Example:
```py
# feature_request/a.py
from typing import Final

class Foo:
    bar: Final = "baz"
```

```py
# feature_request/b.py
from feature_request.a import Foo

print(Foo.bar)
```

Current behaviour:
<img width="259" alt="image" src="https://user-images.githubusercontent.com/11222441/146072151-227962ce-f4c8-4bde-bff0-3c5f9eb20987.png">

Desired behaviour:

<img width="304" alt="image" src="https://user-images.githubusercontent.com/11222441/146072251-c708abef-284b-4935-9847-5ee039cfc81d.png">

<details>
<summary>Workaround for the screenshot</summary>

This is already possible if the source is annotated with both `Literal` and `Final`, but that's redundant if the source was `Final` with a literal assignment right away.

```py
# feature_request/a.py
from typing import Final, Literal

class Foo:
    bar: Final[Literal["baz"]] = "baz"
```

</details>

**Comment by erictraut** [comment]
Thanks for the bug report. I agree that the type should evaluate to a literal in this case. This will be fixed in the next release.

**Comment by zevisert** [comment]
I should note, this applies to non-string types too; I just tested `int` with `bar: Final = 10`. Though if it's just a normal identifier rather than a class property, the literal type shows up.

**Comment by erictraut** [comment]
Yes, my fix applies to all literal types.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.1.0, which we've just released. You can find the changelog here:  [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202210-5-january-2022)
